app/javascript/forms/form-buttons-redux.jsx

```diff
@@ -3,47 +3,12 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import FormButtons from './form-buttons';
+import formButtonsReducer from './form-buttons-reducer';
 
-function initReducer() {
-  if (initReducer.done) {
-    // don't init twice
-    return;
-  }
-  initReducer.done = true;
-
-
-  const initialState = { ...FormButtons.defaultProps };
-
-  const actions = {
-    'FormButtons.init': (_state, payload) => ({ ...initialState, ...payload }),
-    'FormButtons.customLabel': (state, payload) => ({ ...state, customLabel: payload || '' }),
-    'FormButtons.newRecord': (state, payload) => ({ ...state, newRecord: !!payload }),
-    'FormButtons.pristine': (state, payload) => ({ ...state, pristine: !!payload }),
-    'FormButtons.saveable': (state, payload) => ({ ...state, saveable: !!payload }),
-
-    'FormButtons.callbacks': (state, payload) => ({ ...state, ...payload }),
-  };
-
-
-  ManageIQ.redux.addReducer({
-    FormButtons: function FormButtonsReducer(state = initialState, action) {
-      if (actions[action.type]) {
-        return actions[action.type](state, action.payload);
-      } else if (action.type.match(/^FormButtons\./)) {
-        console.warn(`FormButtonsReducer - unknown action type: ${action.type}`, action);
-      }
-
-      return state;
-    },
-  });
-}
-/**
- * This was causing inifinite loop in production ENV
- * We should probably refactor it into proper reducer file and initialize as default store
- */
-initReducer();
+ManageIQ.redux.addReducer({ FormButtons: formButtonsReducer });
 
 function FormButtonsRedux(props) {
+  console.warn('If you are using plain strings form FormButtons action types, please use defined constants from ManageIQ.redux.formButtonsActions to avoid inconsistencies.');
```

Comment:
This shows a popup for every devel when using FormButtonsRedux.

If we can't make the line conditional on actual plain string usage, I'd rather remove it than warn everybody every time.

Reply:
Yeah you are right. There is no way to check if they are using constants. There are some magic ways (middleware) that we can add and listen what actions they are using, but i don't think its a good idea. I will go through the plugins which are using this and replace it there.

Reply:
:+1: 

But, no plugins should be using this at all.

This is currently only used by the provider dialogs feature.

Reply:
nuage uses it. If they use the dispatch action at ManageIQ.redux, we could remove the redux dependency from their repo entirely.